### PR TITLE
chore: move tests from package.json to test.js

### PR DESCRIPTION
**Pull request by Trott**
Remove individual remark tests using the command line from package.json and perform the tests programmatically from test.js instead.

~First commit is https://github.com/nodejs/remark-preset-lint-node/pull/272 which needs to land first.~

### CODE REVIEW

File: test/test.js
Comment (nschonni) [inline test/test.js:28]: Should the file be a little more descriptive and/or follow a pattern? Ex: `bad-title.md`, `bad-mailto.md`, etc...

File: test/test.js
Comment (nschonni) [inline test/test.js:38]: Similar to the other naming comment, maybe `formating-input.md` and `formating-output.md`. I don't see having more than this pair though

File: .eslintignore
Comment (nschonni) [inline .eslintignore]: Shouldn't we still link the test script?

File: package.json
Comment (nschonni) [inline package.json:15]: Shouldn't this call the unit tests somewhere in the chain?

File: .eslintignore
Comment (nschonni) [inline .eslintignore:2]: Shouldn't the test script still be linted?

**Review by nschonni — COMMENTED** [review]
Mostly nitty stuff, but good job!

File: package.json
Comment (nschonni) [reply package.json:15]: Oh, I see, it's being called in the remark.  Maybe we can just drop the `remark` old one, and move the calls here

File: package.json
Comment (nschonni) [inline package.json:12]: Wonder if these should be renamed too. Maybe `test:local-config`, `test:node-core`, etc...

**Comment by nschonni** [comment]
@aduh95 do you have any suggestions for some fixtures for the YAML rule? Doesn't need to be included here though

**Comment by aduh95** [comment]
> @aduh95 do you have any suggestions for some fixtures for the YAML rule? Doesn't need to be included here though

I'll add one after this PR is merged 👍

File: test/test.js
Comment (aduh95) [inline test/test.js:24]: It would be preferable to use TLA here so we can detect if a promise never settles.
```suggestion
await Promise.all([
 (async () => {
  const file = await read(new URL("./fixtures/ok.md", import.meta.url));
  const result = await linter.process(file);
  assert.strictEqual(result.messages.length, 0, reporter(result));
 })(),
 // ...
]);
```

File: test/test.js
Comment (aduh95) [inline test/test.js:2]: nit
```suggestion
import assert from "node:assert";
import fs from "node:fs";
import process from "node:process";
```

**Review by aduh95 — APPROVED** [review]
LGTM modulo Nick's comments

File: test/test.js
Comment (Trott) [reply test/test.js:24]: I originally had TLA for the test but it blew up ESLint and I didn't want to look too deeply. Future improvement?

Comment (aduh95) [reply test/test.js:24]: I suggest adding a comment explaining that. Also, we could re-use the same function for all catch handlers.

File: test/test.js
Comment (Trott) [reply test/test.js:28]: For now, there's only one "make sure it says an OK file is OK" and one "make sure it flags an error as an error". They aren't really checking specific rules or anything. So I'm fine with leaving them as `ok.md` and `fail.md` until we have more of these files. But I also don't feel strongly about it. I just don't know what to change these two files' names to, though.

File: test/test.js
Comment (Trott) [reply test/test.js:38]: Yeah, those are better names. Updating.

File: package.json
Comment (Trott) [reply package.json]: Done.

File: package.json
Comment (Trott) [reply package.json]: Done.

File: test/test.js
Comment (nschonni) [inline test/test.js:36]:
```suggestion
// Test that incorrectly-formatted markdown is turned into correctly-formatted markdown.
```

File: test/test.js
Comment (aduh95) [reply test/test.js:28]: If I'm adding fixtures for YAML in a follow-up PR, we soon will have several test files supposed to fail.

Comment (nschonni) [reply test/test.js:28]: I'm OK if we want to "solve" the naming in that future PR

File: .eslintignore
Comment (Trott) [reply .eslintignore:2]: ESLint can't parse the test file because it uses `import.meta`. We'd need to switch to babel-eslint parser or find a way to avoid `import.meta`. Refs: https://github.com/eslint/eslint/issues/11189#issuecomment-446666614

File: test/test.js
Comment (Trott) [reply test/test.js]: Done.

File: test/test.js
Comment (Trott) [reply test/test.js]: Whoops, fixed.

File: .eslintignore
Comment (Trott) [reply .eslintignore:4]: I put that information in a comment. Optimistically marking this as "resolved" but if you think that's not satisfactory, please unresolve!

File: test/test.js
Comment (Trott) [reply test/test.js:25]: Whoops. The real issue is that TLA is not supported in Node.js 12 and this supports Node.js 12 so that would make adding these tests a breaking change.

Comment (Trott) [reply test/test.js]: > Also, we could re-use the same function for all catch handlers.

Done.

Comment (Trott) [reply test/test.js]: > I suggest adding a comment explaining that.

Comment added. Optimistically marking this as resolved, but unresolve if this resolution does not resolve the issues you think need resolving. Uh, you know what I'm trying to say.

**Comment by Trott** [comment]
@nschonni Does this seem OK to you in its current form? I'm happy to make more changes if you identify things you want improved.

**Review by nschonni — APPROVED** [review]

**Pull request merged**
